### PR TITLE
Add loan vault get and list endpoints

### LINKED ISSUE #418: Loan Vault API
#### What would you like to be added:

Exposing loan vaults API, to be added into `LoanController.ts` at server with `Loan.ts` whale-api-client.

```
GET: /loan/vaults
GET: /loan/vaults/:id
```

/triage accepted
/priority important-soon
/area module-api

### LINKED ISSUE #419: Address Scoped Vault API
#### What would you like to be added:

Exposing address scoped vaults API, to be added into `AddressController.ts` at server with `Address.ts` whale-api-client.

```
GET: /address/:address/vaults
```

/triage accepted
/priority important-soon
/area module-api

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Add loan vault get and list endpoints

#### Which issue(s) does this PR fixes?:

Fixes #418
Fixed #419

### CODE REVIEW

File: packages/whale-api-client/src/api/address.ts [hunk 1]
```diff
@@ -41,6 +42,18 @@ export class Address {
...
 
+  /**
+   * List all vaults belonging to an address.
+   *
+   * @param {string} address bech32/legacy/b58 formatted address
+   * @param {number} size of vaults to query
+   * @param {string} next set of vaults
+   * @return {Promise<ApiPagedResponse<LoanVault>>}
+   */
+  async listVault (address: string, size: number = 30, next?: string): Promise<ApiPagedResponse<LoanVault>> {
```
Comment (canonbrother) [inline packages/whale-api-client/src/api/address.ts:53]:
```suggestion
  async listVaults (address: string, size: number = 30, next?: string): Promise<ApiPagedResponse<LoanVault>> {
```

File: packages/whale-api-client/src/api/loan.ts [hunk 2]
```diff
@@ -61,3 +82,32 @@ export interface CollateralToken {
...
   activateAfterBlock: number
 }
+
+export interface LoanVault {
+  vaultId: string
+  loanSchemeId: string
+  ownerAddress: string
+
+  invalidPrice: boolean
+  isUnderLiquidation: boolean
```
Comment (canonbrother) [inline packages/whale-api-client/src/api/loan.ts]: ohh.. they are replaced by `state` now

File: src/module.api/loan.controller.ts [hunk 3]
```diff
@@ -87,6 +91,28 @@ export class LoanController {
...
   }
+
+  /**
+   * Paginate loan vaults.
+   *
+   * @param {PaginationQuery} query
+   * @return {Promise<ApiPagedResponse<LoanVault>>}
+   */
+  @Get('/vaults')
+  async listVault (@Query() query: PaginationQuery): Promise<ApiPagedResponse<LoanVault>> {
```
Comment (canonbrother) [inline src/module.api/loan.controller.ts:171]:
```suggestion
  async listVaults (@Query() query: PaginationQuery): Promise<ApiPagedResponse<LoanVault>> {
```

File: packages/whale-api-client/__tests__/api/loan.vault.test.ts [hunk 4]
```diff
@@ -67,41 +67,18 @@ afterAll(async () => {
 
 describe('list', () => {
   it('should listVaults with size only', async () => {
-    const result = await client.loanVault.list('', '', 'false', 20)
+    const result = await client.loan.listVault(20)
```
Comment (canonbrother) [inline packages/whale-api-client/__tests__/api/loan.vault.test.ts:70]:
```suggestion
    const result = await client.loan.listVaults(20)
```

File: packages/whale-api-client/src/api/loan.ts [hunk 5]
```diff
@@ -46,6 +46,27 @@ export class Loan {
...
   }
+
+  /**
+   * Paginate query loan vaults.
+   *
+   * @param {number} size of vaults to query
+   * @param {string} next set of vaults
+   * @return {Promise<ApiPagedResponse<LoanVault>>}
+   */
+  async listVault (size: number = 30, next?: string): Promise<ApiPagedResponse<LoanVault>> {
```
Comment (canonbrother) [inline packages/whale-api-client/src/api/loan.ts:79]:
```suggestion
  async listVaults (size: number = 30, next?: string): Promise<ApiPagedResponse<LoanVault>> {
```

File: packages/whale-api-client/src/api/address.ts (hunk 1 above)
Comment (fuxingloh) [reply packages/whale-api-client/src/api/address.ts:53]: I realized plurals it's not properly enforced on the whale project, will take notice of this when we consolidate the design into jellyfish.

Comment (fuxingloh) [reply packages/whale-api-client/src/api/address.ts:53]: Will close these issues for now

File: packages/whale-api-client/__tests__/api/loan.vault.test.ts [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+  try {
+    await service.stop()
+  } finally {
+    await container.stop()
+  }
+})
+
+describe('list', () => {
+  it('should listVault with size only', async () => {
+    // TODO(help-me): make this test more accurate to present the correct structure explicitly
```
Comment (ivan-zynesis) [inline packages/whale-api-client/__tests__/api/loan.vault.test.ts:181]: Have to wait for this, mainly to make sure response of each vault state can be casted correctly.

Haven't check how the data should looks like for UNKNOWN state

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**